tests/test_swift.py

```diff
@@ -29,5 +78,100 @@ def test_frames(self):
         self.assertIsInstance(self.app.frames(), Iterable)
 
 
+class SwiftTest(unittest.TestCase):
+    """Unit test for Swift class in swift.py"""
+
+    def setUp(self):
+        """Create a QApplication and a Swift object every time."""
+        self.qapp = QApplication(sys.argv)
+        importlib.import_module = MagicMock()
+        for appInfo in APP_INFOS.values():
+            app = MagicMock()
+            app.frames.return_value = (QWidget(),)
+            cls = MagicMock(return_value=app)
+            setattr(importlib.import_module.return_value, appInfo.cls, cls)
+        self.buses = set()
+        for appInfo in APP_INFOS.values():
+            self.buses.update(appInfo.bus)
+        self.swift = Swift(APP_INFOS)
+
+    def test_init(self):
+        """Test if swift is initialized correctly."""
+        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
+        self.assertIsInstance(self.swift.centralWidget, QLabel)
+        for name in APP_INFOS:
+            self.assertIn(name, self.swift._apps)
+        for bus in self.buses:
+            self.assertIn(bus, self.swift._subscribers)
+
+    def test_destroy_app(self):
+        """Test destroyApp()."""
+        name = list(APP_INFOS.keys())[0]
+        self.swift.destroyApp(name)
+        self.assertNotIn(name, self.swift._apps)
+
+    def test_broadcast(self):
+        """Test _broadcast()."""
+        busName = list(self.buses)[0]
+        self.swift._broadcast(busName, "test_msg")
+        for app in self.swift._subscribers[busName]:
+            app.received.emit.assert_called_once()
```

Comment:
This seems not clear, because the testing process is not quite predictable or straightforward.
1. `set` iterator would not follow an intuitive order (which bus name will be tested?)
2. What if there is no subscribers for the selected bus?

Therefore, please make it more explicit that which bus name will be tested and how many subscribers there are.

Reply:
> I sorted `self.buses` with alphabetical order.

Is `set` iteration order deterministic? Note that `self.buses` is a `set`, which is an unordered collection.

> I defined `self.buses` as an union set of bus list of `AppInfo`s. So I think there is no case when the bus has no subscribers.

I meant what if `self.swift._subscribers[busName]` is empty?
Then no assertion would be performed, right?
I am worried about this because the code looks like it does not care the tested case (i.e., `busName` and its subscribers here), but it actually does (as you said, you are assuming some properties about the test case).
This may lead to mistakenly hidden bugs.
Consider, later we change the test case, e.g., `APP_INFOS`, and somehow we changed something about `_broadcast()`.
There might be a chance that there is actually a bug, but this test code does not find it, because we changed the test case breaking the assumptions of this test.

Reply:
> Is set iteration order deterministic? Note that self.buses is a set, which is an unordered collection.

I updated `self.buses` as follows.
```python
self.buses = set()
for appInfo in APP_INFOS.values():
    self.buses.update(appInfo.bus)
self.buses = sorted(self.buses)
```
Then, it is changed into a sorted list.